### PR TITLE
Bump buildkit v0.5.0

### PR DESCRIPTION
**Pull request by thaJeztah**
release notes: https://github.com/moby/buildkit/releases/tag/v0.5.0

- https://github.com/moby/buildkit/compare/8818c67cff663befa7b70f21454e340f71616581...v0.5.0
  - moby/buildkit#909 exporter: support unpack opt for image exporter
  - moby/buildkit#961 dockerfile: allow subdirs for remote contexts
- bump containerd/continuity aaeac12a7ffcd198ae25440a9dff125c2e2703a7
  - containerd/continuity#140 Fix directory comparison in changes
- bump containerd 3a3f0aac8819165839a41fee77a4f4ac8b103097
  - full diff: https://github.com/containerd/containerd/compare/ceba56893a76f22cf0126c46d835c80fb3833408...3a3f0aac8819165839a41fee77a4f4ac8b103097
  - https://github.com/containerd/containerd/pull/3241 differ: fix deadlock on commit error 
- bump containerd/console 0650fd9eeb50bab4fc99dceb9f2e14cf58f36e7f
  - containerd/console#30 Add common project repo checks/README references
- bump gogo/protobuf v1.2.1
  - full diff: https://github.com/gogo/protobuf/compare/v1.2.0...v1.2.1
- bump gogo/googleapis v1.2.0
  - full diff: https://github.com/gogo/googleapis/compare/08a7655d27152912db7aaf4f983275eaf8d128ef...v1.2.0
- bump golang.org/x/crypto 88737f569e3a9c7ab309cdc09a07fe7fc87233c3
  - no local changes, just syncing with containerd

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @tonistiigi @AkihiroSuda ptal

**Comment by thaJeztah** [comment]
Ah, dang; need to make some changes

```
22:35:14 Congratulations!  All commits are properly signed with the DCO!
22:36:14 builder/builder-next/adapters/snapshot/snapshot.go:53:34:warning: cannot use &(snapshotter literal) (value of type *snapshotter) as snapshot.SnapshotterBase value in variable declaration: missing method Name (gosimple)
22:36:14 builder/builder-next/adapters/snapshot/snapshot.go:74:9:warning: cannot use s (variable of type *snapshotter) as snapshot.SnapshotterBase value in return statement: missing method Name (unconvert)
22:36:14 builder/builder-next/adapters/snapshot/snapshot.go:53:34:warning: cannot use &(snapshotter literal) (value of type *snapshotter) as snapshot.SnapshotterBase value in variable declaration: missing method Name (unconvert)
22:36:15 Build step 'Execute shell' marked build as failure
```

File: builder/builder-next/adapters/snapshot/snapshot.go [hunk 1]
```diff
@@ -74,6 +74,10 @@ func NewSnapshotter(opt Opt) (snapshot.SnapshotterBase, error) {
 	return s, nil
 }
 
+func (s *snapshotter) Name() string {
+	return "default"
```
Comment (thaJeztah) [inline builder/builder-next/adapters/snapshot/snapshot.go]: Wasn't entirely sure where the name is used, and what to name this one, so let me know what I should use here

**Comment by tonistiigi** [comment]
buildkit changes LGTM, don't know about the containerd/proto parts

**Comment by dmcgowan** [comment]
LGTM

File: builder/builder-next/adapters/snapshot/snapshot.go (hunk 1 above)
Comment (AkihiroSuda) [reply builder/builder-next/adapters/snapshot/snapshot.go]: @tonistiigi WDYT? `moby` seems more consistent?

Comment (tonistiigi) [reply builder/builder-next/adapters/snapshot/snapshot.go]: It is not used anywhere in moby so doesn't matter much. I'm fine with either.

**Review by tiborvass — APPROVED** [review]

**Pull request merged**